engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/TestProcess.scala

```diff
@@ -49,14 +49,22 @@ object TestProcess {
   /*
     We have to be careful not to put too much into results, as they are serialized to JSON.
    */
-  case class NodeResult[+T](context: ResultContext[T])
+  case class NodeResult[T](context: ResultContext[T]) {
 
-  case class ExpressionInvocationResult[+T](contextId: String, name: String, value: T)
+    def variableTyped[U <: T](name: String): Option[U] = context.variableTyped(name)
```

Comment:
Take a look at usages in `TransformersTest`. It looks like we have some common pattern, that we use `TestResults[Any]` in our processes tests and we want to downcast variables to more specific type. If you add such boundary :`[U <: T]` it is impossible to have covariant type `+T`, but it looks like we haven't used this covariance and was added "just in case". The second option is to add some utils for tests that will do such downcast in one place but I'm not the fun of complex matrix of utils. WDYT?

Reply:
ok, I think I understand ;)